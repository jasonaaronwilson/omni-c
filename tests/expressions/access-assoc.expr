// This commit gets it wrong:
// dd65554df92cc7ae52a9194fdaa18d6dca39480d
//
// (token_list.(list->length))
//
// Luckily we can see this is kind of dumb since token_list.NUMBER
// wouldn't make sense and we know if real life this code works.

token_list.list->length;
