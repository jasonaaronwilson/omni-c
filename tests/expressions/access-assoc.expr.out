{
    tag = PARSE_NODE_OPERATOR
    operator = ->
    left = {
        tag = PARSE_NODE_OPERATOR
        operator = .
        left = {
            tag = PARSE_NODE_IDENTIFIER
            token = token_list
        }
        right = {
            tag = PARSE_NODE_IDENTIFIER
            token = list
        }
    }
    right = {
        tag = PARSE_NODE_IDENTIFIER
        token = length
    }
}

// C Output
((token_list.list)->length)
